server/api/audit_boards.py

```diff
@@ -83,6 +84,21 @@ def assign_sampled_ballots(
             db_session.add(ballot)
 
 
+def assign_sampled_batches(
+    jurisdiction: Jurisdiction, round: Round, audit_boards: List[AuditBoard]
```

Comment:
what would be the advantage of feeding in the `db_session`? are you worried that it doesn't occur in the same transaction otherwise? my understanding is that SQLAlchemy creates a new transaction per thread, so when using Flask, all calls to the global `db_session` per request are part of one transaction that is committed when you call `commit`, and then a new transaction begins the next time you make a call to `db_session`. But maybe I've got it wrong

Reply:
more context:
- based our design on this pattern: https://flask.palletsprojects.com/en/1.1.x/patterns/sqlalchemy/#declarative
- sqlalchemy docs has a lot more details on this approach here: https://docs.sqlalchemy.org/en/13/orm/contextual.html#using-thread-local-scope-with-web-applications

Reply:
it does feel a bit confusing though. i think it could be nice to make it more explicit (maybe explicitly instantiate a session transaction at the beginning of each request?)

Reply:
to be clear, I don't think there's anything wrong with what you're doing in the actual code, you're committing at the top-level. But it requires a bit of implicit knowledge, when you write a subprocedure, that it should never `.commit()`. But also, why should it never `.commit()`? What is the rule for where you call `.commit()`?

I was just wondering if having a rule that says "if you're not planning on committing your changes, then take a `db_session` as a parameter to make that decision clear. It's clearly uncool to `.commit()` on a parameter that's passed to you, so that is, in a way, a mild forcing function for consistent behavior.

I'm not dead-set on this, though. Just wondering about long-term maintenance of this code and how to make the implicit practices explicit.

Reply:
oh ok cool you are getting at something else. i love trying to make the code more readable/maintainable, so appreciate this suggestion

> It's clearly uncool to .commit() on a parameter that's passed to you

Is this a known convention? It's not something I've ever encountered before

Reply:
(merging this PR because this conversation expands beyond the scope of the PR, but let's keep talking because I can always revisit in a separate PR)